sympy/solvers/tests/test_solveset.py

```diff
@@ -809,39 +809,46 @@ def test_solve_lambert():
 
 
 def test_solveset():
-    x = Symbol('x', real=True)
+    x = Symbol('x')
     raises(ValueError, lambda: solveset(x + y))
 
-    assert solveset(exp(x) - 1) == FiniteSet(0)
-    assert solveset(exp(x) - 1, x) == FiniteSet(0)
-    assert solveset(Eq(exp(x), 1), x) == FiniteSet(0)
+    assert solveset(exp(x) - 1, domain=S.Reals) == FiniteSet(0)
+    assert solveset(exp(x) - 1, x, S.Reals) == FiniteSet(0)
+    assert solveset(Eq(exp(x), 1), x, S.Reals) == FiniteSet(0)
 
-    assert solveset(x - 1 >= 0, x) == Interval(1, oo)
-    assert solveset(exp(x) - 1 >= 0, x) == Interval(0, oo)
+    assert solveset(x - 1 >= 0, x, S.Reals) == Interval(1, oo)
+    assert solveset(exp(x) - 1 >= 0, x, S.Reals) == Interval(0, oo)
 
-    x = Symbol('x')
     assert solveset(exp(x) - 1, x) == imageset(Lambda(n, 2*I*pi*n), S.Integers)
     assert solveset(Eq(exp(x), 1), x) == imageset(Lambda(n, 2*I*pi*n),
                                                   S.Integers)
 
 
+def test_solveset_domain():
+    x = Symbol('x')
+
+    assert solveset(exp(x) - 1, x, S.Reals) == FiniteSet(0)
```

Comment:
Repeated test as [this](https://github.com/sympy/sympy/pull/9762/files#diff-85baa04bbf4e1dfd9128782738e45424R815)